prov/sockets/include/sock.h

```diff
@@ -870,6 +870,10 @@ struct sock_conn_hdr {
 	uint8_t reserved[3];
 	uint16_t port;
 	uint16_t cm_data_sz;
+};
+
+struct sock_conn_resp {
+	struct sock_conn_hdr hdr;
```

Comment:
The code isn't actually using this structure though.  There are places where sock_conn_hdr is used where it would be clearer using sock_conn_resp.  I think we should perform those updates as well.  Otherwise, we're better off just commenting out cm_data from sock_conn_hdr.